### ISSUE TITLE
Line number and column number information

### ISSUE THREAD

**Issue by Rameshkumar1990** [stleary/JSON-java#366]
Do we have a facility to track Line number or Column number of the Nodes in JSONObject or JSONTokener?

**Comment by johnjaylward** [comment]
The JSONTokener has a tracker in it for line and column positions when parsing. The JSONObject only tracks that when parsing an initial string into itself using the JSONTokener.

https://github.com/stleary/JSON-java/blob/master/JSONTokener.java#L42

The `character` and `line` are the variables that track the position. They are then used when generating a [syntaxError](https://github.com/stleary/JSON-java/blob/master/JSONTokener.java#L504)

**Comment by migueltt** [comment]
@Rameshkumar1990  Note that those attributes @johnjaylward is referring to, are private.
The only way to know exactly where (row, col) the error is located would be analyzing (parsing) the JSONException message (which is thrown only if the JSON-string is not well-defined).
TL;DR: No, you cannot track line, column - you can only get an error message with the location.

**Comment by stleary** [comment]
There are no plans to include line and column information in the API. Feel free to make a pull request if you can think of a way to do this in a way that does not significantly impact performance, and which does not modify the existing API.
